src/serializer/types.js

```diff
@@ -19,6 +19,14 @@ import invariant from "../invariant.js";
 
 export type TryQuery<T> = (f: () => T, defaultValue: T, logFailures: boolean) => T;
 
+// TODO: add type for additional functions.
+export type EmitBodyType = "MainGenerator" | "Generator" | "DelayInitialization" | "Other";
+
+export type EmitBody = {
```

Comment:
I don't think `EmitBody` is best in terms of english grammar. If `Body` itself is not available, I'd prefer `SerializedBody`, or maybe `CodeBody`.